TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -201,10 +208,11 @@ public void onServiceDisconnected(ComponentName componentName) {
     };
 
     @Override
-    public void onConnect(String room) {
+    public void onConnect(String room, String serverAddress) {
         debugView.showPermanently(getString(R.string.connected));
         switchableView.setDisplayedChild(1);
 
+        lastServerPreferences.saveLastConnectedServer(serverAddress);
```

Comment:
Good points @blundell, @Mecharyry.
I extracted an interface for the server address persistence and moved the two save invocations to the presenters (adding tests).
Have a look at https://github.com/novoda/spikes/pull/299/commits/2db5c21bd9c98ae19a6ddea1f1ff3b8580ef3ac9 and https://github.com/novoda/spikes/pull/299/commits/7d467488933c7ddaf9155a2f4f516a4d91046713